cmd/k8s-operator/ports.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand/v2"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"tailscale.com/kube/kubetypes"
+)
+
+const (
+	portMax            = 65535
+	portMin            = 1024
+	testSvcName     = "test-node-port-range"
+
+	invalidSvcNodePort = 777777
+)
+
+var kubePortRangeErr = fmt.Sprintf("Service \"%s\" is invalid: spec.ports[0].nodePort: Invalid value: %d: provided port is not in the valid range. The range of valid ports is ", invalidSvcName, invalidSvcNodePort)
+
+type portRange struct {
+	Start  int
```

Comment:
I think we can use `uint16` for port values

Reply:
Yeah we are gonna move to `tailcfg.PortRange` - in the process of doing this now.